app/models/rental.rb

```diff
@@ -168,6 +167,12 @@ def delete_reservation(uuid)
     return false
   end
 
+  # same as delete_reservations but it aborts the destroy
+  def before_destroy_delete_reservations
+    delete_reservations # will add errors to obj
+    throw(:abort) if errors.any? # abort a #destroy
```

Comment:
Would it be possible to use a [transaction](http://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html) instead?

Reply:
I don't think they are interchangeable.  The throw abort is to abort a ActiveRecord destroy action.  However now that I think of it, I think rails won't even complete the destroy unless errors are empty.

Reply:
jsut checked and it is necessary to explicitly say `throw(:abort)`, just having the errors won't do anything.